### DISCUSSION TITLE
Mender on Xavier-NX using the wip-jetpack-5-dp branch

### DISCUSSION THREAD

**Discussion by f-gia** [OE4T/meta-tegra#1006] [Q&A] [answered]
Hello.

We were trying to integrate Mender support in our image for the Xavier-NX emmc devkit, and added the `meta-mender-tegra` layer (which relies on the `dunfell-l4t-r32.4.3` branch of `meta-tegra`) to our build. However, we are using the `wip-jetpack-5-dp` branch of `meta-tegra`, in order to get a more recent kernel version.

We noticed that in this branch, the u-boot and c-boot recipes disappeared, so that bitbake complains about missing recipes for the bbappend files in `meta-mender-tegra`. We were wondering whether integration of a jetpack-5-based system with Mender is possible and whether it is planned for the future.

Thanks for your support.

**Comment by ichergui** [comment, marked as answer]
Hello @f-gia 
I don't think we have the `A/B system`  update via Mender working in our branch `wip-jetpack-5-dp`
We are still working on the BSP side 
Let's see if anyone have another idea

**Reply by kekiefer** [reply]
You beat me to it. The new BSP uses UEFI to boot and not cboot or u-boot. UEFI should support the A/B, but the runtime tegra bootloader upgrade tools still need to modified to work with the new scheme.

**Reply by dwalkes** [reply]
I'm still curious to see how NVIDIA is going to handle the Jetpack 4 to Jetpack 5 transition.  It will be one thing to port mender support over to support Jetpack 5 to Jetpack 5 upgrades, and another thing to support Jetpack 4.6 to 5 transitions.  AFAIK they haven't released any support for Jetpack 4 to 5 upgrades yet.

**Reply by kekiefer** [reply]
Yes I too would like to see how "[this feature will be supported in next public release](https://docs.nvidia.com/jetson/archives/r34.1/DeveloperGuide/text/SD/SoftwarePackagesAndTheUpdateMechanism.html?highlight=ota#updating-jetson-linux-with-image-based-over-the-air-update)"

**Comment by hannes09** [comment]
Hi, any news for supporting mender with jetpack 5.0 ?  Has anyone tried yet?

**Reply by dwalkes** [reply]
I don't know that anyone has tried yet.  I think it would be possible now but note that NVIDIA support for A/B is [not yet ready](https://docs.nvidia.com/jetson/archives/r35.1/DeveloperGuide/text/SD/SoftwarePackagesAndTheUpdateMechanism.html#trigger-the-ota-process).  There are some messages on gitter about success booting a/b partitions by hacking UEFI variables by hand - see https://gitter.im/OE4T/jetpack-5-test?at=6309262172ad51741fa1d436 for instance.  So it's probably possible to start now based on this.  I don't know that anyone is actively working on this yet, however.

**Reply by dwalkes** [reply]
We discussed this briefly in the monthly meeting this month as well - see https://github.com/OE4T/meta-tegra/wiki/OE4T-Meeting-Notes-2022-10-13

**Reply by dwalkes** [reply]
Related thread also at https://github.com/OE4T/tegra-demo-distro/discussions/231

**Reply by dwalkes** [reply]
Related thread at https://matrix.to/#/!hOxPRrEdLCqPlLQeuE:gitter.im/$gDjQijO1QgGqg-3wOTJkjzmgGEHdmgTxmL1y-1-9ulY?via=gitter.im&via=matrix.org
Discussed in the monthly meetings at https://github.com/OE4T/meta-tegra/wiki/OE4T-Meeting-Notes-2023-01-12 and https://github.com/OE4T/meta-tegra/wiki/OE4T-Meeting-Notes-2023-02-09
